app/controllers/api/custom_buttons_controller.rb

```diff
@@ -2,29 +2,27 @@ module Api
   class CustomButtonsController < BaseController
     def create_resource(_type, _id, data)
       CustomButton.transaction do
-        custom_button = CustomButton.new(data.except("resource_action", "options"))
-        custom_button.userid = User.current_user.userid
-        custom_button.options = data["options"].deep_symbolize_keys if data["options"]
-        custom_button.resource_action.update!(data["resource_action"].deep_symbolize_keys) if data.key?("resource_action")
-        custom_button.save!
-        custom_button
+        CustomButton.new(data.except("resource_action", "options")).tap do |cb|
+          cb.userid = User.current_user.userid
+          cb.options = data["options"].deep_symbolize_keys if data["options"]
+          cb.resource_action = ResourceAction.create!(data["resource_action"].deep_symbolize_keys) if data.key?("resource_action")
+          cb.save!
+        end
       end
     rescue
-      raise BadRequestError, "Failed to create new custom button - #{custom_button.errors.full_messages.join(", ")}"
+      raise BadRequestError, "Failed to create new custom button - #{$!}"
```

Comment:
Instead of `!$` you can `rescue => error` and use `error.message` here

Reply:
done in the last commit before you commented